api/proto/teleport/legacy/types/types.proto

```diff
@@ -3665,6 +3665,10 @@ message KubernetesResource {
   string Name = 3 [(gogoproto.jsontag) = "name,omitempty"];
   // Verbs are the allowed Kubernetes verbs for the following resource.
   repeated string Verbs = 4 [(gogoproto.jsontag) = "verbs,omitempty"];
+
+  // Group specifies the API group of the resource. Can be '*' for any,
+  // but doesn't support pattern matching.
+  string Group = 5 [(gogoproto.jsontag) = "group,omitempty"];
```

Comment:
```suggestion
  string ApiGroup = 5 [(gogoproto.jsontag) = "group,omitempty"];
```

I am also in favor of adding subresources for future compatibility

Reply:
Done in #54418